src/background/services/storage.ts

```diff
@@ -31,10 +32,25 @@ const defaultStorage = {
 } satisfies Omit<Storage, 'publicKey' | 'privateKey' | 'keyId'>
 
 export class StorageService {
+  private setRemainingBalanceRecurring: DebounceWithQueue<[amount: string]>
+  private setRemainingBalanceOneTime: DebounceWithQueue<[amount: string]>
+
   constructor(
     private browser: Browser,
     private events: EventsService
-  ) {}
+  ) {
+    const bigIntMin = (a: string, b: string) => (BigInt(a) < BigInt(b) ? a : b)
+    this.setRemainingBalanceRecurring = new DebounceWithQueue(
+      (amount: string) => this.set({ recurringGrantRemainingBalance: amount }),
+      (args) => [args.reduce((m, [e]) => (!m ? e : bigIntMin(m, e)), '')],
```

Comment:
If we're getting multiple calls to `Storage.setRemainingBalance` (which maybe out of order depending on which createOutgoingPayment resolves first), we can collect the calls and update storage less often.
It might be ok to update balance in storage on each createOutgoingPayment also - maybe just unnecessary optimization, but I think more meaningful when we have like 5-6+ wallet addresses per page.